process_jartic_archives.py

```
@@ -0,0 +1,280 @@
+    if output_path.exists():
+        print(f"File size: {output_path.stat().st_size / 1024 / 1024:.1f} MB")
+    print(f"{'='*60}")
+    
+    return output_path, total_measurements
+
+
+def parse_date(date_str):
+    try:
+        return datetime.strptime(date_str, '%Y-%m-%d').replace(tzinfo=timezone.utc)
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"Invalid date format: '{date_str}'. Use YYYY-MM-DD")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)